### PR TITLE
Add abs2() operator for squared abs()

### PR DESCRIPTION
**Pull request by tbensonatl**
Add abs2() operator that is equivalent to abs()*abs() without the extra loads and the unnecessary sqrt operators associated with complex abs() calculations.

### CODE REVIEW

File: docs_input/api/math/misc/abs2.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.. _abs2_func:
+
+abs2
+====
+
+Squared absolute value. For complex numbers, this is the squared
+complex magnitude, or real(t)^2 + imag(t)^2. For real numbers,
```
Comment (cliffburdick) [inline docs_input/api/math/misc/abs2.rst:7]: If you want to get fancy in rst you can use `\ :sup:`2``

File: test/00_operators/OperatorTests.cu [hunk 2]
```diff
@@ -1582,6 +1582,73 @@ TYPED_TEST(OperatorTestsAllExecs, OperatorFuncs)
...
+TYPED_TEST(OperatorTestsNumericAllExecs, Abs2)
+{
+  MATX_ENTER_HANDLER();
+  using TestType = std::tuple_element_t<0, TypeParam>;
+  using ExecType = std::tuple_element_t<1, TypeParam>;
+  using inner_type = typename inner_op_type_t<TestType>::type;
+
+  ExecType exec{};
+
+  auto sync = [&exec]() constexpr {
```
Comment (cliffburdick) [inline test/00_operators/OperatorTests.cu:1594]: We should probably have a sync function in the executor at this point

**Review by cliffburdick — APPROVED** [review]

File: test/00_operators/OperatorTests.cu (hunk 2 above)
Comment (tbensonatl) [reply test/00_operators/OperatorTests.cu:1594]: Agreed. I can create a PR to add that. I think we only use this lambda sync pattern in the unit tests currently.

**Pull request merged**